### PR TITLE
KAZOO-1243: correctly pass outbound flags and filter ratedeck's based…

### PR DESCRIPTION
**Pull request by mark2600**
… on them as well as options

### CODE REVIEW

File: applications/callflow/src/module/cf_resources.erl [hunk 1]
```diff
@@ -92,6 +92,7 @@ build_offnet_request(Data, Call) ->
                             ,{<<"Bypass-E164">>, get_bypass_e164(Data)}
                             ,{<<"Inception">>, get_inception(Call)}
                             ,{<<"B-Leg-Events">>, [<<"DTMF">>]}
+			    ,{<<"Custom-Channel-Vars">>, {[{<<"Authorizing-ID">>, whapps_call:authorizing_id(Call)}]}}
```
Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_resources.erl]: Want to nudge this over one more space? :)

File: applications/ecallmgr/src/ecallmgr_fs_authz.erl [hunk 2]
```diff
@@ -192,11 +192,16 @@ set_ccv_trunk_usage(JObj, CallId, Node) ->
...
-    Type = wh_json:get_value(<<"Account-Billing">>, JObj),
+    Type      = wh_json:get_value(<<"Account-Billing">>, JObj),
+    ChanVars  = wh_json:get_value(<<"Custom-Channel-Vars">>, JObj),
+
     lager:debug("call authorized by account ~s as ~s", [AccountId, Type]),
-    P = props:set_values([{?GET_CCV(<<"Account-ID">>), AccountId}
-                          ,{?GET_CCV(<<"Account-Billing">>), Type}
-                         ], Props),
+    P = props:set_values([
+	{?GET_CCV(<<"Account-ID">>), AccountId}
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_authz.erl]: Preferred to drop the open list down:

```erlang
props:set_value(
    [{?GET_CCV(...), AccountId}
     ,{?GET_CCV(...), ...}
    ]
)
```

File: applications/ecallmgr/src/ecallmgr_fs_authz.erl [hunk 3]
```diff
@@ -372,5 +377,6 @@ rating_req(CallId, Props) ->
      ,{<<"Account-ID">>, kzd_freeswitch:account_id(Props)}
      ,{<<"Direction">>, kzd_freeswitch:call_direction(Props)}
      ,{<<"Send-Empty">>, 'true'}
+     ,{<<"Outbound-Flags">>, proplists:get_value(<<"Outbound-Flags">>, Props)}
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_authz.erl]: Use `props:get_value/2`

File: applications/callflow/src/module/cf_resources.erl (hunk 1 above)
Comment (lazedo) [inline applications/callflow/src/module/cf_resources.erl]: move this to get_channel_vars as we may want to set more vars in the future.

`{<<"Custom-Channel-Vars">>, get_channel_vars(Call) }`

```
get_channel_vars(Call) ->
   wh_json:from_list(
     [{<<"Authorizing-ID">>, whapps_call:authorizing_id(Call)}]
   ).
```

File: applications/jonny5/src/j5_authz_req.erl [hunk 4]
```diff
@@ -208,23 +208,33 @@ maybe_inbound_soft_limit(Request, Limits) ->
...
-              ,{<<"Reseller-Billing">>, j5_request:reseller_billing(Request)}
-              ,{<<"Call-Direction">>, j5_request:call_direction(Request)}
-              ,{<<"Other-Leg-Call-ID">>, j5_request:other_leg_call_id(Request)}
-              ,{<<"Soft-Limit">>, wh_util:to_binary(j5_request:soft_limit(Request))}
-              ,{<<"Msg-ID">>, j5_request:message_id(Request)}
-              ,{<<"Call-ID">>, j5_request:call_id(Request)}
-              ,{<<"Custom-Channel-Vars">>, CCVs}
-              | wh_api:default_headers(?APP_NAME, ?APP_VERSION)
-             ]),
+    Account  = wh_util:format_account_id(j5_request:account_id(Request), 'encoded'),
```
Comment (jamesaimonetti) [inline applications/jonny5/src/j5_authz_req.erl]: Use `AccountDb` for the name so readers know what encoding it is in

File: applications/jonny5/src/j5_request.erl [hunk 5]
```diff
@@ -124,6 +127,9 @@ request_number(Number, CCVs) ->
             Original
     end.
 
+ccvs(Request) ->
+    Request#request.request_ccvs.
```
Comment (jamesaimonetti) [inline applications/jonny5/src/j5_request.erl]: My personal preference is to pattern match records in the header vs accessing them in the body, if possible. I'm not sure what the performance implications are, if the compiler converts both to the same Core Erlang, or what, but just throwing it out there :)

File: applications/callflow/src/module/cf_resources.erl (hunk 1 above)
Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_resources.erl]: +1 good call @lazedo

**Comment by jamesaimonetti** [comment]
@mark2600 I fixed the spec in cf_resources for get_channel_vars/1 and the list formatting.

**Pull request merged**